geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/BaseCommand.java

```diff
@@ -1463,12 +1464,20 @@ protected static Operation getOperation(final Part operationPart,
     }
 
     // Fallback for older clients.
-    final Operation operation = (Operation) operationPart.getObject();
+    final Object operation = operationPart.getObject();
     if (operation == null) {
-      // native clients may send a null since the op is java-serialized.
+      // older native clients may send a null since the op was java-serialized.
       return defaultOperation;
     }
-    return operation;
-  }
 
+    if (operation instanceof Byte) {
```

Comment:
I think this static method should be deleted and Put65/Destroy65 should have their own implementations of getOperation().  Then you could have this one-off case handled in the Destroy65 version of the method.